### PR TITLE
Fix ruby-end hook removal

### PR DESCRIPTION
**Pull request by TheBB**
Fixes https://github.com/syl20bnr/spacemacs/issues/4059

### CODE REVIEW

File: layers/+lang/elixir/packages.el [hunk 1]
```diff
@@ -130,10 +130,15 @@
         (set (make-variable-buffer-local 'ruby-end-check-statement-modifiers)
              nil)
         (ruby-end-mode +1))
-      (add-hook 'elixir-mode-hook 'spacemacs//ruby-end))
+      (add-hook 'elixir-mode-hook 'spacemacs//ruby-end)
+      ;; hack to remove the autoloaded `add-hook' in `ruby-end'
+      ;; since they are inserted as an autoload, they have to be removed both
+      ;; before and after loading
```
Comment (aaronjensen) [inline layers/+lang/elixir/packages.el]: For my own understanding, why does them being inserted as an autoload necessitate removal in both config and init?

Comment (TheBB) [inline layers/+lang/elixir/packages.el]: Autoloaded code runs before everything.

**Comment by robbyoconnor** [comment]
:heart:

**Comment by TheBB** [comment]
Cherry-picked.

**Pull request closed without merging**